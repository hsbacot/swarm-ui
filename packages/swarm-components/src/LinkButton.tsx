import * as React from 'react';
import { Icon } from './Icon';
import { ButtonProps } from './Button';
import { getButtonType, getSwarmSize, getIconPosition } from './utils/buttonUtils';

export interface LinkButtonProps extends ButtonProps {};

const LinkButton = (props: LinkButtonProps): React.AnchorHTMLAttributes<LinkButtonProps> => {
	// destructuring to not pass invalid attributes to node
	const {
		icon,
		right,
		grow,
		children,
		bordered, // eslint-disable-line no-unused-vars
		neutral, // eslint-disable-line no-unused-vars
		primary, // eslint-disable-line no-unused-vars
		inverted, // eslint-disable-line no-unused-vars
		small, // eslint-disable-line no-unused-vars
		...other
	} = props;

	const buttonType = getButtonType(props);
	const width = grow ? 'grow' : 'default';

	return (
		<a
			data-swarm-link={buttonType}
			data-swarm-size={getSwarmSize(props)}
			data-icon={getIconPosition(props)}
			data-swarm-width={width}
			{...other}
		>
			{icon ? (
				<span>
					{right ? (
						<>
							{children}
							<Icon shape={icon} size="xs" />
						</>
					) : (
						<>
							<Icon shape={icon} size="xs" />
							{children}
						</>
					)}
				</span>
			) : (
				children
			)}
		</a>
	);
};

export { LinkButton };
