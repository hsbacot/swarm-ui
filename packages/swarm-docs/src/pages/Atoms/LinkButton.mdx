import { LinkButton } from '@meetup/swarm-components';
import { Icon } from '@meetup/swarm-components';
import Example from '../../components/examples/linkButton';
import PropsTable from '../../components/propstable';

## LinkButton
---
An html anchor tag styled as a button.

<br/>
<br/>

### Snippet
---

Use this interactive snippet to try the component:
<br/>

```.jsx
<LinkButton href="https://www.meetup.com" primary>Link styled as Button</LinkButton>
```
<br/>
<br/>

### Props
---
<br/>
<PropsTable propMetaData={LinkButton.__docgenInfo} />

<br/>
<br/>

### a11y
---
Keyboard interactions: The button action should be triggered by the enter key.

<br/>
<br/>

## Examples
---

<Example />
