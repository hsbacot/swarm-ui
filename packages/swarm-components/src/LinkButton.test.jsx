import React from 'react';
import { shallow } from 'enzyme';
import { LinkButton } from './LinkButton';

describe('LinkButton', () => {
	const testCases = [
		['Default', <LinkButton key="1">Press me</LinkButton>],
		[
			'Disabled',
			<LinkButton disabled key="2">
				Can&apos;t press me
			</LinkButton>,
		],
		[
			'Primary',
			<LinkButton primary key="3">
				Must press me
			</LinkButton>,
		],
		[
			'Bordered',
			<LinkButton bordered key="4">
				Press me
			</LinkButton>,
		],
		[
			'Icon with text',
			<LinkButton icon="bolt" key="5">
				Press me
			</LinkButton>,
		],
		[
			'Icon (right) with text',
			<LinkButton right icon="chevron-right" key="6">
				Press me
			</LinkButton>,
		],
		['Icon without text', <LinkButton icon="bolt" key="7" />],
		[
			'Inverted',
			<LinkButton inverted key="8">
				Press me
			</LinkButton>,
		],
		[
			'Neutral',
			<LinkButton neutral key="9">
				Press me
			</LinkButton>,
		],
		[
			'Small',
			<LinkButton small key="10">
				Press me
			</LinkButton>,
		],
		[
			'Grow',
			<LinkButton grow key="11">
				Full Width Button
			</LinkButton>,
		],
	];
	test.each(testCases)('Snapshot: %s', (description, element) => {
		expect(shallow(element)).toMatchSnapshot();
	});
});
