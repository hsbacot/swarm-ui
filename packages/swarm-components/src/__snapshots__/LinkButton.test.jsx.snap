// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton Snapshot: Bordered 1`] = `
<a
  data-icon="left"
  data-swarm-link="bordered"
  data-swarm-size="default"
  data-swarm-width="default"
>
  Press me
</a>
`;

exports[`LinkButton Snapshot: Default 1`] = `
<a
  data-icon="left"
  data-swarm-link="default"
  data-swarm-size="default"
  data-swarm-width="default"
>
  Press me
</a>
`;

exports[`LinkButton Snapshot: Disabled 1`] = `
<a
  data-icon="left"
  data-swarm-link="disabled"
  data-swarm-size="default"
  data-swarm-width="default"
  disabled={true}
>
  Can't press me
</a>
`;

exports[`LinkButton Snapshot: Grow 1`] = `
<a
  data-icon="left"
  data-swarm-link="default"
  data-swarm-size="default"
  data-swarm-width="grow"
>
  Full Width Button
</a>
`;

exports[`LinkButton Snapshot: Icon (right) with text 1`] = `
<a
  data-icon="right"
  data-swarm-link="default"
  data-swarm-size="default"
  data-swarm-width="default"
>
  <span>
    Press me
    <Icon
      shape="chevron-right"
    />
  </span>
</a>
`;

exports[`LinkButton Snapshot: Icon with text 1`] = `
<a
  data-icon="left"
  data-swarm-link="default"
  data-swarm-size="default"
  data-swarm-width="default"
>
  <span>
    <Icon
      shape="bolt"
    />
    Press me
  </span>
</a>
`;

exports[`LinkButton Snapshot: Icon without text 1`] = `
<a
  data-icon="only"
  data-swarm-link="default"
  data-swarm-size="default"
  data-swarm-width="default"
>
  <span>
    <Icon
      shape="bolt"
    />
  </span>
</a>
`;

exports[`LinkButton Snapshot: Inverted 1`] = `
<a
  data-icon="left"
  data-swarm-link="inverted"
  data-swarm-size="default"
  data-swarm-width="default"
>
  Press me
</a>
`;

exports[`LinkButton Snapshot: Neutral 1`] = `
<a
  data-icon="left"
  data-swarm-link="neutral"
  data-swarm-size="default"
  data-swarm-width="default"
>
  Press me
</a>
`;

exports[`LinkButton Snapshot: Primary 1`] = `
<a
  data-icon="left"
  data-swarm-link="primary"
  data-swarm-size="default"
  data-swarm-width="default"
>
  Must press me
</a>
`;

exports[`LinkButton Snapshot: Small 1`] = `
<a
  data-icon="left"
  data-swarm-link="default"
  data-swarm-size="small"
  data-swarm-width="default"
>
  Press me
</a>
`;
